store/store.go

```diff
@@ -670,7 +673,7 @@ func (s *SnapUbuntuStoreRepository) Download(remoteSnap *snap.Info, pbar progres
 	if err != nil {
 		return "", err
 	}
-	s.setUbuntuStoreHeaders(req, "", auther)
+	s.setUbuntuStoreHeaders(req, "", remoteSnap.NeedsDevMode(), auther)
```

Comment:
at this point, yes, sending `x-ubuntu-confinement:devmode` is exactly what we need to do. We got the `remoteSnap` from the store by asking for it according to the user's instructions, and if that remote snap has `confinement:devmode`, we need to set the header again.